frame/treasury/src/lib.rs

```diff
@@ -163,10 +163,24 @@ pub trait Trait: frame_system::Trait {
 /// An index of a proposal. Just a `u32`.
 pub type ProposalIndex = u32;
 
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug)]
+pub enum ProposalStatus<BlockNumber> {
+	/// Proposal is pending approval.
+	Pending,
+	/// Proposal is approved for a one time payout.
+	Approved(BlockNumber),
+	/// Proposal is approved for a reoccurring payout.
+	/// First `BlockNumber` is the next payout block.
+	/// Second `BlockNumber` is the period between payouts.
+	/// `u32` represents the number of payouts left.
+	Reoccurring(BlockNumber, BlockNumber, u32),
```

Comment:
Would need to actually integrate this option if we went down this path

Reply:
```suggestion
	Recurring(BlockNumber, BlockNumber, u32),
```